src/ukify/ukify.py

```diff
@@ -587,6 +601,25 @@ def pe_add_sections(uki: UKI, output: str):
 
     pe.write(output)
 
+def merge_sbat(input: [pathlib.Path]) -> str:
+    sbat = []
+
+    for f in input:
+        try:
+            pe = pefile.PE(f, fast_load=True)
+        except pefile.PEFormatError:
+            print(f"{f} is not a valid PE file, not extracting SBAT section.")
+            continue
+
+        for section in pe.sections:
+            if section.Name.rstrip(b"\x00").decode() == ".sbat":
+                split = section.get_data().rstrip(b"\x00").decode().splitlines()
+                # Filter out the sbat line, we'll add it back later, there needs to be only one and it
+                # needs to be first.
+                sbat += [line for line in split if not line.startswith('sbat,')]
+
+    return 'sbat,1,SBAT Version,sbat,1,https://github.com/rhboot/shim/blob/main/SBAT.md\n' + '\n'.join(sbat) + "\n\x00"
```

Comment:
and then a function join_sbat() or so that undoes parse_sbat() and prepends the line, basically doing what this line does?

i mean, i am just not a fan of too adhoc parsers and compilers stuffed into overly long lines like this one...